rfcs/0182-user-management.md

```diff
@@ -0,0 +1,306 @@
+their configurations.
+
+```shell
+$> ### help message, possibly pointing to further detailed and/or specialized help, like `--all`, `--modules`
+$> hometool help
+$> ### generate a sample config; can be enhanced with a wizard
+$> hometool sample-config > user-configuration.nix
+$> ### because we like to tweak things a bit
+$> emacs user-configuration.nix
+$> ### build it first without deploying it yet
+$> hometool build
+$> ### a container to test it before deploying - for us paranoids!
+$> hometool container
+$> ### now install it!
+$> hometool install
+$> ### list the generations
+$> hometool generations list
+$> ### list differences between generations
+$> hometool generations diff 9 10
+$> ### select a specific generation
+$> hometool generations select 10
+$> ### remove older generations, marking them to be garbage-collected
+$> hometool generations remove 1-8
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Why not to keep this tool outside Nixpkgs monorepo?
+
+  It can be argued that overlays and other facilities provided by Nix language
+  allow to keep entire projects outside the Nixpkgs monorepo.
+
+  However, any of those projects suffer of an unpleasant phenomenon: the second
+  class citizen repo.
+
+  Basically, a repository outside the monorepo will not receive the same level
+  of care and attention of a project inside the monorepo. It happens because,
+  among other things,
+
+  - it is harder to pay attention in multiple repositories at the same time slot
+    than only one;
+  - it is harder to deal with pieces of code contained in distinct repositories;
+  - it is harder to synchronize multiple repositories;
+
+  Because of those hurdles, it becomes easier to ignore the repos outside the
+  main monorepo.
+
+  As a consequence, breaking changes in the monorepo are prone to be decided
+  without taking the satellite repos into consideration. This situation leads
+  the satellite repos to deal with the extra work of synchronizing with the
+  recent changes.
+
+  By living inside the main monorepo, the problems exposed above diminishes
+  drastically.
+
+  Further, having this home management toolset inside the monorepo brings many
+  advantages:
+
+  - No longer dealing with synchronization among projects
+  - Better strategies for code deduplication, sharing and refactoring
+  - Direct access to the most recent enhancements available
+
+    Indeed this is precisely what happens in NixOS already: when a program is
+    added, a NixOS module and test suite can be plugged in the same pull
+    request, with few to no bureaucracy.
+
+  - Reputation
+
+    A home management toolset kept in the monorepo has the social add-on of
+    being better regarded by the Nixpkgs community as a whole, since there will
+    be no barriers for contribution from this same community.
+
+- The monorepo will increase in size.
+
+  The increase in size is expected to be small.
```

Comment:
Well, my estimation is way worse in this prospect. 
If the current home-manager is only 20% of the current nixos size, then it is less than 10% of 20% - just 2%.

Reply:
So the idea is that this will provide the same functionality as Home Manager but with a tenth of the code? How did you derive that number?

Reply:
> So the idea is that this will provide the same functionality as Home Manager but with a tenth of the code? 

No.
I am working on the heuristic/guess of "in the worst case, this hometool moduleset should have a similar size of the already existing NixOS folder", or "hometool == nixos".
Another possibility would be "hometool == home-manager" - and home-manager is considerably smaller than NixOS.

> How did you derive that number?

From what getchoo provided.

Not sure if I understood your question here.